node/test/go/dev-mode/test/integration/index.test.js

```diff
@@ -81,6 +82,10 @@ describe('Integration', function () {
               // Since we are sending spans as they are complete I need to change this to a variable length
               // but we should always have at least 1
               expect(traces.length >= 1).to.equal(true);
+              // For whatever reason the traces are being written in a bad wire format? 🤔
+              // const traceData = Buffer.from(traces[0], 'base64');
+              // const tracePayload = traceProto.TracePayload.decode(traceData);
```

Comment:
Not sure exactly what's `traces` here, but in counterpart in Node.js testing, here you receive already decoded Trace, which is decoded in `process-function` util, in this way: https://github.com/serverless/console/blob/8008eb90bd7c8af375bd830b448666043f484643/node/packages/aws-lambda-sdk/test/lib/process-function.js#L254

Reply:
In this context, `traces` is an array of base64 trace strings. I believe what I am doing here is equivalent to what is happening above it just doesn't what to translate the base64 string to a valid proto object for some reason. I tried adding the `.trim()` but that doesn't seem to be the issue either.

Reply:
I plan to refactor out `process-function` from `aws-lambda-sdk` to top `node` folder (so we can also rely on it for Python SDK testing) hopefully tomorrow. I think then it'll be good to update this test to that as well.

If it's not issue in test utils config, it would mean that written report is for some reason broken

Reply:
Yeah something is up with this test or build or something because I just [tried forcing the integration tests to run on github in a separate branch](https://github.com/serverless/console/actions/runs/3952735645/jobs/6768192872) and those failed with a similar wire format error 😢 

I guess I am still technically covered on this update without decoding the trace payload and explicitly checking since we are getting the correct number of logs. (If this wasn't working there would be 3 instead of 1) It is of course just less explicit.

Reply:
Ahh I am dumb 🤦 I just realized the issue. We are sending trace payloads to the dev mode extension wrapped as a [DevMode Payload](https://github.com/serverless/console/blob/main/proto/serverless/instrumentation/v1/dev_mode.proto) to add additional context. So this is not a `TracePayload` it is `dev_mode` payload.

Totally forgot we were doing that lol I will update these test to reflect this 👍